### PR TITLE
Fixes task reconnection status does not work for default build task

### LINKED ISSUE #159639: Task reconnection status does not work for default build task
Recording 2022-08-31 at 08 20 47

I urge to selfhost on this for a bit. Status bar never shows the state for me, I think never has after window reload.

### LINKED ISSUE #164807: Terminal restored itself after hiding/closing
* hide/close terminal
* reload window
* terminal is back

https://user-images.githubusercontent.com/1794099/198307011-79e73f2f-c037-413b-b1fe-c1e3e13629f8.mov

This is likely caused by my autorun-task but it set to never reveal itself

```
{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "npm",
			"script": "watch",
			"group": "build",
			"problemMatcher": [],
			"label": "npm: watch",
			"detail": "npx esbuild index.ts --outfile=index.js --watch",
			"presentation": {
				"echo": true,
				"reveal": "never",
				"focus": false,
				"panel": "shared",
				"showReuseMessage": true,
				"clear": false
			},
			"runOptions": {
				"runOn": "folderOpen"
			}
		}
	]
}

```

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #159639
Fixes #164807
And some :broom: :ok_hand:

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 1]
```diff
@@ -374,10 +373,8 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			return;
 		}
 		this._getTaskSystem();
-		this._waitForSupportedExecutions.then(() => {
-			this.getWorkspaceTasks().then(async () => {
```
Comment (jeanp413) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:378]: `getWorkspaceTasks` already awaits `_waitForSupportedExecutions` internally

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 2]
```diff
@@ -220,7 +220,6 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 
 	protected _outputChannel: IOutputChannel;
 	protected readonly _onDidStateChange: Emitter<ITaskEvent>;
-	protected readonly _onDidReconnectToTerminals: Emitter<void> = new Emitter();
```
Comment (jeanp413) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:223]: not used anymore

**Comment by meganrogge** [comment]
Thanks for looking into this, i'll check it out tomorrow 👍🏼

**Review by meganrogge — COMMENTED** [review]
Works well

**Review by meganrogge — APPROVED** [review]

**Review by alexr00 — APPROVED** [review]

**Pull request merged**